static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx

```diff
@@ -27,28 +35,90 @@ function useOnboardingRouter(initialStep: OnboardingStep): OnboardingRouterState
       // Add the edges between the old and the new step
       current.next = next;
       next.previous = current;
-      // Return the neext step
+      // Return the next step
       return next;
     });
   }, []);
 
   return [state, toStep];
 }
 
-interface ProfilingOnboardingModalProps extends ModalRenderProps {}
-export function ProfilingOnboardingModal(props: ProfilingOnboardingModalProps) {
+// The wrapper component for all of the onboarding steps. Keeps track of the current step
+// and all state. This ensures that moving from step to step does not require users to redo their actions
+// and each step can just re-initialize with the values that the user has already selected.
+export function ProfilingOnboardingModal(props: ModalRenderProps) {
   const [state, toStep] = useOnboardingRouter({
     previous: null,
     current: SelectProjectStep,
     next: null,
   });
-  return <state.current {...props} toStep={toStep} step={state} />;
+  const [project, setProject] = useState<Project | null>(null);
+
+  return (
+    <state.current
+      {...props}
+      toStep={toStep}
+      step={state}
+      project={project}
+      setProject={setProject}
+    />
+  );
+}
+
+const SUPPORTED_PLATFORMS = new Set(['android', 'apple-ios']);
+function isProfilingSupportedFor(project: Project): boolean {
+  if (!project.platform) {
+    return false;
+  }
+  return SUPPORTED_PLATFORMS.has(project.platform);
```

Comment:
should this use the `platformToInstructionMapping` to determine support instead of having to update 2 consts independently?

Reply:
Yeah, lets do that